kale/prepdata/tabular_transform.py

```diff
@@ -0,0 +1,87 @@
+# =============================================================================
+# Author: Sina Tabakhi, sina.tabakhi@gmail.com
+# =============================================================================
+
+"""
+Provide data transformations for tabular datasets.
+"""
+
+from typing import Any, Optional
+
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+
+class ToTensor(object):
+    r"""Convert an array_like data to a tensor of the same shape. This class provides a callable object that allows
+    instances of the class to be called as a function. In other words, this class wraps the functionality of
+    `torch.tensor <https://pytorch.org/docs/stable/generated/torch.tensor.html>`__ and allows users to use it as a
+    callable instance.
+
+    Args:
+        dtype (torch.dtype, optional): The desired data type of returned tensor. Default: if :obj:`None`, infers data
+            type from data.
+        device (torch.device, optional): The device of the constructed tensor. If :obj:`None` and data is a tensor then
+            the device of data is used. If None and data is not a tensor then the result tensor is constructed on the
+            CPU.
+    """
+
+    def __init__(self, dtype: Optional[torch.dtype] = None, device: Optional[torch.device] = None) -> None:
+        self.dtype = dtype
+        self.device = device
+
+    def __call__(self, data: Any) -> Tensor:
+        """
+        Args:
+            data (array_like): Initial data for the tensor. Can be a list, tuple, NumPy ndarray, scalar, and other
+                types.
+
+        Returns:
+            Data converted to tensor.
+        """
+        return torch.tensor(data, dtype=self.dtype, device=self.device)
+
+
+class ToOneHotEncoding(object):
+    r"""Convert an array_like of class values of shape ``(*,)`` to a tensor of shape ``(*, num_classes)`` that have
```

Comment:
@XianyuanLiu Can you please take a look at this and let me know your opinion? Thanks.

Reply:
[sklearn.preprocessing.OneHotEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html#sklearn-preprocessing-onehotencoder) seems the similar function. Please check if the output is callable. If not, this function is needed.

Reply:
@shuo-zhou Add on here.

Reply:
https://pytorch.org/docs/stable/generated/torch.nn.functional.one_hot.html